src/lib.rs

```diff
@@ -71,15 +71,60 @@ impl WaiterChannel {
     }
 }
 
-struct Scheduler {
-    waiter: Arc<WaiterChannel>,
-    heap: BinaryHeap<Schedule>,
+/// A trait that allows configurable execution of scheduled item
+/// on the scheduler thread.
+trait Executor<T> {
+    fn execute(&mut self, data : T);
```

Comment:
Hmm... Do you mean the Executor trait itself or the execute/execute_clone distinction (which I agree is unfortunate)?